TicTacToe/TicTacToe.js

```diff
@@ -0,0 +1,157 @@
+            if (this.user == 'X') {
+                return 'O';
+            } else {
+                return 'X';
+            }
+        }
+
+        this.board[x][y] = this.user;
+        this.moves++;
+
+        if (this.isWin(x, y)) {
+            this.messages = this.user + " is the winner!";
+            this.complete = true;
+            return this.user;
+        }
+
+        if (this.moves == 9) {
+            this.messages = "It's a tie!";
+            return this.user;
+        }
+
+        if (this.user == 'X') {
+            this.user = 'O';
+            return 'X';
+        } else {
+            this.user = 'X';
+            return 'O';
+        }
+    };
+
+    isWin(x, y) {
+        let win1 = true;
+        let win2 = true;
+        let win3 = true;
+        let win4 = true;
+
+        // check row
+        for (let i = 0; i < 3; i++) {
+            if (this.board[x][i] != this.user) {
+                win1 = false;
+            }
+        }
+        // check col
+        for (let i = 0; i < 3; i++) {
+            if (this.board[i][y] != this.user) {
+                win2 = false;
+            }
+        }
+
+        // check diags
+        for (let i = 0; i < 3; i++) {
+            if (this.board[i][i] != this.user) {
+                win3 = false;
+            }
+        }
+
+        for (let i = 0; i < 3; i++) {
+            if (this.board[i][2 - i] != this.user) {
+                win4 = false;
+            }
+        }
+
+        return win1 || win2 || win3 || win4;
+    };
+
+}
+
+let game = new Game();
+let moves = document.getElementById("moves");
+let messages = document.getElementById("messages");
+let reset = document.getElementById("newGame");
+let buttons = new Array();
+makeButtons();
+buttons.forEach(element => element.innerHTML = ".");
+
+function makeButtons() {
+    for (let i = 0; i < 3; i++) {
+        for (let j = 0; j < 3; j++) {
+            let button = document.getElementById(i + "" + j);
+            button.addEventListener('click', function () {
+                if (game.complete) {
+                    return;
+                }
+                buttons.push(button);
+
+                let m = game.makeMove(i, j);
+
+                button.innerHTML = m;
+                messages.innerHTML = "Messages: " + game.getMessages();
+                debug.innerHTML = game.displayBoard();
+                moves.innerHTML = "Move count: " + game.getMoves();
+            });
+        }
+    }
+}
+
+function start() {
+    console.log("New Game Started!");
+    game.reset();
+    console.log(game.complete);
+    document.getElementById("debug").innerHTML = game.displayBoard();
+    moves.innerHTML = "Move count: " + game.getMoves();
+    messages.innerHTML = "Messages: " + game.getMessages();
+    buttons.forEach(element => element.innerHTML = ".");
+}
+
+// why does this only work when I use "function()"?
+reset.addEventListener('click', function () {
+    start();
+});
+
+
+start();
```

Comment:
All in all, your coding skill has definitely improved a lot this semester (year?). This code is coherent and pretty straightforward, and is fairly good JS.

Reply:
<(￣︶￣)>